at_secondary/at_secondary_server/lib/src/verb/handler/scan_verb_handler.dart

```diff
@@ -107,42 +111,59 @@ class ScanVerbHandler extends AbstractVerbHandler {
   }
 
   /// Returns the current atsign keys.
-  /// @param - atConnectionMetadata: Metadata of the inbound connection
-  /// @param - List<String>: List of keys from the secondary persistent store
-  /// @return - String: Returns the keys of current atsign
-  String _getLocalKeys(
-      InboundConnectionMetadata atConnectionMetadata, List<String?> keys) {
-    var keyString;
+  ///
+  /// **Parameters**
+  ///
+  /// [atConnectionMetadata] Metadata of the inbound connection.
+  ///
+  /// [keys] List of keys from the secondary persistent store.
+  ///
+  /// **Returns**
+  ///
+  /// Returns the list of keys of current atsign.
+  List<String> _getLocalKeys(
+      InboundConnectionMetadata atConnectionMetadata, List<String> keys) {
+    List<String> keysList = [];
     // Verify if the current user is authenticated or not
     // If authenticated get all the keys except for private keys
     // If not, get only public keys
     if (atConnectionMetadata.isAuthenticated) {
       //display all keys except private
       keys.removeWhere((key) =>
-          key.toString().startsWith('privatekey:') ||
-          key.toString().startsWith('public:_') ||
-          key.toString().startsWith('private:'));
-      keyString = keys.toString();
+          _isPrivateKeyForAtSign(key, 'privatekey:') ||
```

Comment:
please change this to _isPrivateKeyForAtSign(key)..the checks can happen inside the method impl

Reply:
Yup done